packages/store/addon/-private/system/store.ts

```diff
@@ -273,16 +299,17 @@ const Store = Service.extend({
     @private
     @return Adapter
   */
-  defaultAdapter: computed('adapter', function() {
-    let adapter = get(this, 'adapter');
+  @computed('adapter')
+  get defaultAdapter() {
+    let adapter = (this as any).adapter || '-json-api';
```

Comment:
I found a pattern that lets us indicate the presence of an `adapter` field on the class as _purely type information_, to avoid babel treating it (correctly) as a class field w/o an initializer

```ts
interface C {
    noInit: string;
}

class C {
  constructor() {
      this.noInit // string
  }
}

export { C };
```
b/c of declaration merging, TS combines `class C` and `interface C`, but babel will completely ignore the `interface`